spec.emu

```diff
@@ -276,6 +276,95 @@
       <emu-note>That means that "A" (`LATIN CAPITAL LETTER A`) measures as 1 code unit, and "🔥" (`FIRE`) measures as 2 code units. Source maps for other content types may diverge from this.</emu-note>
     </dd>
   </dl>
+
+  <emu-clause id="sec-vlq-grammar">
+    <h1>base64 VLQ grammar</h1>
+    <p>A base64-encoded variable-length quantity adheres to the following lexical grammar:</p>
+
+    <emu-grammar type="definition">
+      Vlq ::
+        VlqDigitList
+
+      VlqDigitList ::
+        TerminalDigit
+        ContinuationDigit VlqDigitList
+
+      TerminalDigit ::
+        OneOfTerminalDigit
+
+      ContinuationDigit ::
+        OneOfContinuationDigit
+
+      OneOfTerminalDigit :: one of
+        `A` `B` `C` `D` `E` `F` `G` `H` `I` `J` `K` `L` `M` `N` `O` `P` `Q` `R`
+        `S` `T` `U` `V` `W` `X` `Y` `Z` `a` `b` `c` `d` `e` `f`
+
+      OneOfContinuationDigit :: one of
+        `g` `h` `i` `j` `k` `l` `m` `n` `o` `p` `q` `r` `s` `t` `u` `v` `w` `x`
+        `y` `z` `0` `1` `2` `3` `4` `5` `6` `7` `8` `9` `+` `/`
+    </emu-grammar>
+
+    <emu-note type="editor">
+      |OneOfTerminalDigit| and |OneOfContinuationDigit| will go away and are only present in the preview draft. They are here because ecmarkup currently does not support *one of* productions in SDOs.
+    </emu-note>
+
+    <emu-clause id="sec-VLQSignedValue" type="sdo">
+      <h1>VLQSignedValue ( ): an integer</h1>
+      <dl class="header"></dl>
+      <emu-grammar>
+        Vlq :: VlqDigitList
+      </emu-grammar>
+      <emu-alg>
+        1. Let _unsigned_ be the VLQUnsignedValue of |VlqDigitList|.
+        1. If _unsigned_ modulo 2 = 1, let _sign_ be -1.
+        1. Else, let _sign_ be 1.
+        1. Let _value_ be floor(_unsigned_ / 2).
+        1. If _value_ is 0 and _sign_ is -1, return -2<sup>31</sup>.
+        1. If _value_ is ≥ 2<sup>31</sup>, throw an error.
```

Comment:
I would maybe add a note somewhere that we need this check because we are calling VLQUnsignedValue on |VlqDigitList| rather than on |Vlq|, so we are skipping VLQUnsignedValue's validation.

Or we call VLQUnsignedValue on |Vlq|, and replace this check with an assertion.

Reply:
Added a note. Otherwise we'd have to turn `VLQSignedValue` into an abstract op and pass in a |Vlq| parse node.